validator/accounts/v2/wallet.go

```diff
@@ -300,32 +301,59 @@ func (w *Wallet) ReadPasswordForAccount(accountName string) (string, error) {
 }
 
 func (w *Wallet) enterPasswordForAccount(cliCtx *cli.Context, accountName string) error {
-	attemptingPassword := true
-	// Loop asking for the password until the user enters it correctly.
-	for attemptingPassword {
-		// Ask the user for the password to their account.
-		password, err := inputPasswordForAccount(cliCtx, accountName)
+	var password string
+	var err error
+	if cliCtx.IsSet(flags.PasswordFileFlag.Name) {
+		passwordFilePath := cliCtx.String(flags.PasswordFileFlag.Name)
+		data, err := ioutil.ReadFile(passwordFilePath)
 		if err != nil {
-			return errors.Wrap(err, "could not input password")
+			return err
 		}
-		accountKeystore, err := w.keystoreForAccount(accountName)
-		if err != nil {
-			return errors.Wrap(err, "could not get keystore")
-		}
-		decryptor := keystorev4.New()
-		_, err = decryptor.Decrypt(accountKeystore.Crypto, []byte(password))
+		password = string(data)
+		err = w.checkPasswordForAccount(accountName, password)
 		if err != nil && strings.Contains(err.Error(), "invalid checksum") {
-			fmt.Println("Incorrect password entered, please try again")
-			continue
+			return fmt.Errorf("invalid password entered for account %s", accountName)
 		}
 		if err != nil {
-			return errors.Wrap(err, "could not decrypt keystore")
+			return err
 		}
+	} else {
+		attemptingPassword := true
+		// Loop asking for the password until the user enters it correctly.
+		for attemptingPassword {
+			// Ask the user for the password to their account.
+			password, err = inputPasswordForAccount(cliCtx, accountName)
+			if err != nil {
+				return errors.Wrap(err, "could not input password")
+			}
+			err = w.checkPasswordForAccount(accountName, password)
+			if err != nil && strings.Contains(err.Error(), "invalid checksum") {
```

Comment:
Maybe this invalid checksum err message can be a package var from the direct keymanager package, something ErrPasswordDoesNotMatch

Reply:
This error comes from the keystore library unfortunately, not from code we maintain.